Automattic/jetpack
# Check currently stored value of the `mfold` user setting before updating

The `set_user_setting` function always updates the settings update timestamp, making the stored value unique, which leads to DB writes and user meta cache being purged on every wp-admin screen.

In this changeset, a logic for checking the currently stored value is added, which allows the code to only call the `set_user_setting` function if the value needs to be updated, ultimatelly preventing unneccessary database writes, and user meta cache purging.





Fixes #25332

#### Changes proposed in this Pull Request:

*

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Have memcache or other persitant cache storage installed in your testing environment
* Have a debug bar, query monitor, or some other plugin allowing to inspect SQL queries for a request installed
* Have Gutenberg the plugin installed
* Go to the /wp-admin screen
* Check the SQL queries, a SQL query for collecting and updating all user meta should be present
* Reload the /wp-admin screen, the same SQL query should be still present
* Apply the patch repeat the 2 previous steps, the SQL query should not repeat itself
